frappe/public/js/legacy/clientscriptAPI.js

```diff
@@ -248,7 +248,16 @@ _f.Frm.prototype.clear_table = function(fieldname) {
 _f.Frm.prototype.add_child = function(fieldname, values) {
 	var doc = frappe.model.add_child(this.doc, frappe.meta.get_docfield(this.doctype, fieldname).options, fieldname);
 	if(values) {
-		$.extend(doc, values);
+		// Deep clone
+		var d = JSON.parse(JSON.stringify(values));
```

Comment:
instead of this, just loop over the keys and copy the values, excluding the blacklist